src/stan_math_backend/Statement_gen.ml

```diff
@@ -75,9 +69,34 @@ let trans_math_fn fname =
     value ~default:(fname, [])
       (bind (internal_fn_of_string fname) ~f:math_fn_translations))
 
+let rec block_wrapping {stmt; smeta} =
+  let in_block {stmt; smeta} =
+    { stmt=
+        ( match stmt with
+        | Block l -> Block l
+        | SList l -> Block l
```

Comment:
Aren't SLists different semantically from Blocks?

Reply:
Yes, but we only call block_wrapping on individual statement, not on lists of statements. Then, we have that {S} and S have exactly the same semantics, as long as we have a single statement. The goal is only to be able to add a second statement updating the location anywhere where we have a single statement. For example, if we start with
```
if 2 >3 
  print(foo);
```
then, to transform it to the appropriate location update, we need to add a block:
```
if 2>3 {
  loc = 4;
  print(foo);
}
```